ShaiRandom/Generators/KnownSeriesRandom.cs

```diff
@@ -730,88 +745,75 @@ public IEnhancedRandom StringDeserialize(ReadOnlySpan<char> data)
             _intIndex = int.Parse(data.Slice(idx + 1, -1 - idx + (idx = data.IndexOf('~', idx + 1))));
             _intSeries.Clear();
             var seriesData = data.Slice(idx + 1, -1 - idx + (idx = data.IndexOf('~', idx + 1)));
-            if (!seriesData.IsEmpty)
-            {
-                foreach (var numData in seriesData.Tokenize('|'))
+            foreach (var numData in seriesData.Tokenize('|'))
+                if (!numData.IsEmpty)
                     _intSeries.Add(int.Parse(numData));
-            }
+
             // UInt
             _uintIndex = int.Parse(data.Slice(idx + 1, -1 - idx + (idx = data.IndexOf('~', idx + 1))));
             _uintSeries.Clear();
             seriesData = data.Slice(idx + 1, -1 - idx + (idx = data.IndexOf('~', idx + 1)));
-            if (!seriesData.IsEmpty)
-            {
-                foreach (var numData in seriesData.Tokenize('|'))
+            foreach (var numData in seriesData.Tokenize('|'))
+                if (!numData.IsEmpty)
```

Comment:
I like this better than what I had before. It gives an extra defense against a potential `||` showing up, though I don't know how it would.